packages/styled-base/types/index.d.ts

```diff
@@ -53,41 +53,68 @@ export interface StyledComponent<InnerProps, StyleProps, Theme extends object>
 }
 
 type ReactClassPropKeys = keyof React.ClassAttributes<any>
-export interface CreateStyledComponentBase<
+export type CreateStyledComponentBase<
```

Comment:
Using `type` instead of `interface` will expose internal types instead of hide details.
For types like this (or types that are complex enough), using `type` will give you complex information when you trying to see the type in editors.

Reply:
interfaces cant be conditional though, that's why I've switched to type

Reply:
How about using two interfaces for each conditional cases to hide complex overloadings?
I mean, just like
```
interface A { ... }
interface B { ... }
type CreateStyledComponentBase =
  cond ?
    A :
    B;
```

Reply:
Out of curiosity - could you specify when exactly (preferably with a simple demo using TS playground) this matters? From my experience interfaces & types are interchangeable nowadays, but maybe I haven't noticed smth.

Reply:
@Andarist As far as I remember,
```
// file1
import { A } from "something" // A is type
import { B } from "otherthing" // B is interface

const x: A = ...
const y: B = ...

export { x, y }
```

```
// file2
import { x, y } from "file1"

x // IDE will give you internal definition of A
y // IDE will give you the name of B
```

I'm not sure it's still the case. If it's not, please let me know, I will be glad to hear that!

Reply:
I think this has changed, I often see a type alias instead of its content - anyhow we might use interface here as well, doesnt matter (for me). I've amended the PR.